# Interacting Dummy and Regular Variables (First Example)

### Intro and objectives

### In this lab you will learn:
1. examples of simple regression models with dummy variables interacting with regular ones.
2. how to fit simple regression models in Python.


## What I hope you'll get out of this lab
* The feeling that you'll "know where to start" when you need to fit a simple regression model and include several dummy variables interacting with regular ones.
* Examples of simple regression models
* How to interpret the results obtained

In [1]:
!pip install wooldridge
import wooldridge as woo
import statsmodels.formula.api as smf
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.1/5.1 MB 20.1 MB/s eta 0:00:00


# Example 1. Log Hourly Wage, marital status and gender

## In this example we want to test for differences in return to education accross gender.

### Using the data in WAGE1 where n=526 individuals

In [2]:
Wages = woo.dataWoo('wage1')


In [3]:
Wages.head()

wage  educ  exper  tenure  nonwhite  female  married  numdep  smsa  \
0  3.10    11      2       0         0       1        0       2     1   
1  3.24    12     22       2         0       1        1       3     1   
2  3.00    11      2       0         0       0        0       2     0   
3  6.00     8     44      28         0       0        1       0     1   
4  5.30    12      7       2         0       0        1       1     0   

   northcen  ...  trcommpu  trade  services  profserv  profocc  clerocc  \
0         0  ...         0      0         0         0        0        0   
1         0  ...         0      0         1         0        0        0   
2         0  ...         0      1         0         0        0        0   
3         0  ...         0      0         0         0        0        1   
4         0  ...         0      0         0         0        0        0   

   servocc     lwage  expersq  tenursq  
0        0  1.131402        4        0  
1        1  1.175573      484        4  
2        0  1.098612        4        0  
3        0  1.791759     1936      784  
4        0  1.667707       49        4  

[5 rows x 24 columns]

In [4]:
Wages.describe()

wage        educ      exper      tenure    nonwhite      female  \
count  526.000000  526.000000  526.00000  526.000000  526.000000  526.000000   
mean     5.896103   12.562738   17.01711    5.104563    0.102662    0.479087   
std      3.693086    2.769022   13.57216    7.224462    0.303805    0.500038   
min      0.530000    0.000000    1.00000    0.000000    0.000000    0.000000   
25%      3.330000   12.000000    5.00000    0.000000    0.000000    0.000000   
50%      4.650000   12.000000   13.50000    2.000000    0.000000    0.000000   
75%      6.880000   14.000000   26.00000    7.000000    0.000000    1.000000   
max     24.980000   18.000000   51.00000   44.000000    1.000000    1.000000   

          married      numdep        smsa    northcen  ...    trcommpu  \
count  526.000000  526.000000  526.000000  526.000000  ...  526.000000   
mean     0.608365    1.043726    0.722433    0.250951  ...    0.043726   
std      0.488580    1.261891    0.448225    0.433973  ...    0.204680   
min      0.000000    0.000000    0.000000    0.000000  ...    0.000000   
25%      0.000000    0.000000    0.000000    0.000000  ...    0.000000   
50%      1.000000    1.000000    1.000000    0.000000  ...    0.000000   
75%      1.000000    2.000000    1.000000    0.750000  ...    0.000000   
max      1.000000    6.000000    1.000000    1.000000  ...    1.000000   

            trade    services    profserv     profocc     clerocc     servocc  \
count  526.000000  526.000000  526.000000  526.000000  526.000000  526.000000   
mean     0.287072    0.100760    0.258555    0.366920    0.167300    0.140684   
std      0.452826    0.301298    0.438257    0.482423    0.373599    0.348027   
min      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
50%      0.000000    0.000000    0.000000    0.000000    0.000000    0.000000   
75%      1.000000    0.000000    1.000000    1.000000    0.000000    0.000000   
max      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   

            lwage      expersq      tenursq  
count  526.000000   526.000000   526.000000  
mean     1.623268   473.435361    78.150190  
std      0.531538   616.044772   199.434664  
min     -0.634878     1.000000     0.000000  
25%      1.202972    25.000000     0.000000  
50%      1.536867   182.500000     4.000000  
75%      1.928619   676.000000    49.000000  
max      3.218076  2601.000000  1936.000000  

[8 rows x 24 columns]

In [5]:
type(Wages)

pandas.core.frame.DataFrame

In [6]:
Wages.columns

Index(['wage', 'educ', 'exper', 'tenure', 'nonwhite', 'female', 'married',
       'numdep', 'smsa', 'northcen', 'south', 'west', 'construc', 'ndurman',
       'trcommpu', 'trade', 'services', 'profserv', 'profocc', 'clerocc',
       'servocc', 'lwage', 'expersq', 'tenursq'],
      dtype='object')

## Let's interact female with educ:

In [7]:
# We impose a simple, linear, model:
# We specify CeoSalaries as the empirical dataset

reg = smf.ols(formula='np.log(wage) ~female+educ+female*educ+exper+np.power(exper,2)+tenure+np.power(tenure,2)', data=Wages)

In [8]:
# We fit the model
results = reg.fit()


In [9]:
b = results.params
print(f'b: \n{b}\n')

b: 
Intercept              0.388806
female                -0.226789
educ                   0.082369
female:educ           -0.005565
exper                  0.029337
np.power(exper, 2)    -0.000580
tenure                 0.031897
np.power(tenure, 2)   -0.000590
dtype: float64



In [10]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:           np.log(wage)   R-squared:                       0.441
Model:                            OLS   Adj. R-squared:                  0.433
Method:                 Least Squares   F-statistic:                     58.37
Date:                Thu, 01 Aug 2024   Prob (F-statistic):           1.67e-61
Time:                        09:46:12   Log-Likelihood:                -260.49
No. Observations:                 526   AIC:                             537.0
Df Residuals:                     518   BIC:                             571.1
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                          coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               0.3888    

## How do we interpret the equation?

#### The R-squared in relatively large (R-squared=0.44).
#### The F-statistic is large (58.37) with p-value close to zero. Therefore the model is statistically significant.

#### R-squared is 0.44 (therefore 44% of the variance is explained by this model)

#### Education, experience and tenureship are statistically significant.

#### Female is not statistically significant (p-value>0.05). We observe than once we control for experience and tenureship, wages are not influenced by gender.

#### The interaction term female*education is not statistically significant (p-value>0.05). this means that the financial return of education is the same for males and females.

#### Both experience and tenureship exhibit quadratic behaviour (diminishing effects given that their coefficients are negative, -0.0006 and -0.0006 respectively).






